backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/KotlinObjCClassInfoGenerator.kt

```diff
@@ -27,10 +27,17 @@ internal class KotlinObjCClassInfoGenerator(override val context: Context) : Con
 
         val instanceMethods = generateInstanceMethodDescs(irClass)
 
-        val companionObject = irClass.declarations.filterIsInstance<IrClass>().atMostOne { it.isCompanion  }
+        val companionObject = irClass.companionObject()
         val classMethods = companionObject?.generateMethodDescs().orEmpty()
 
-        val superclassName = irClass.getSuperClassNotAny()!!.let {
+        val superClass = irClass.getSuperClassNotAny()!!
+        if (superClass.isObjCClass() && companionObject != null) {
+            val companionBodySize = LLVMStoreSizeOfType(llvmTargetData, context.llvmDeclarations.forClass(companionObject).bodyType).toInt()
+            if (companionBodySize > LLVMStoreSizeOfType(llvmTargetData, int8TypePtr).toInt())
+                context.reportCompilationError("Fields are not supported for Companion of subclass of ObjC type")
```

Comment:
We have bunch of compilation errors reported from `InteropLowering.kt`.
I guess this check should be moved there too.